File: fix_errors.py
#!/usr/bin/env python3
"""
Script to fix common errors in the codebase:
1. Fix invalid characters (backslashes before quotes)
2. Resolve merge conflicts
3. Fix basic syntax errors
"""

import os
import re
import glob
from pathlib import Path

def fix_invalid_characters(content):
    """Fix invalid characters like \' to '"""
    # Fix backslashes before single quotes
    content = re.sub(r"\\'", "'", content)
    # Fix backslashes before double quotes
    content = re.sub(r'\\"', '"', content)
    return content

def resolve_merge_conflicts(content):
    """Resolve merge conflicts by keeping the HEAD version"""
    # Remove merge conflict markers and keep HEAD version
    lines = content.split('\n')
    result = []
    in_conflict = False
    keep_lines = True
    
    for line in lines:
<<<<<<< HEAD
        if line.strip().startswith(''):
=======
        if line.strip().startswith('            in_conflict = True
            keep_lines = True
            continue
        elif line.strip().startswith(''):
>>>>>>> origin/cursor/fix-errors-and-merge-to-main-34b5
            keep_lines = False
            continue
        elif line.strip().startswith('>>>>>>>'):
            in_conflict = False
            keep_lines = True
            continue
        elif not in_conflict or keep_lines:
            result.append(line)
    
    return '\n'.join(result)

def fix_jsx_syntax(content):
    """Fix common JSX syntax errors"""
    # Fix missing closing tags in self-closing elements
    content = re.sub(r'<(\w+)([^>]*?)(?<!/)>', r'<\1\2 />', content)
    
    # Fix malformed className attributes
    content = re.sub(r'className="([^"]*?)\s*pt-20"', r'className="\1 pt-20"', content)
    
    return content

def fix_typescript_imports(content):
    """Fix TypeScript import issues"""
    # Fix 'use client' directive placement
    if "'use client';" in content:
        lines = content.split('\n')
        if "'use client';" in lines:
            lines.remove("'use client';")
            lines.insert(0, "'use client';")
        content = '\n'.join(lines)
    
    return content

def process_file(file_path):
    """Process a single file to fix errors"""
    try:
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()
        
        original_content = content
        
        # Apply fixes
        content = fix_invalid_characters(content)
        content = resolve_merge_conflicts(content)
        content = fix_jsx_syntax(content)
        content = fix_typescript_imports(content)
        
        # Only write if content changed
        if content != original_content:
            with open(file_path, 'w', encoding='utf-8') as f:
                f.write(content)
            print(f"Fixed: {file_path}")
            return True
        else:
            print(f"No changes needed: {file_path}")
            return False
            
    except Exception as e:
        print(f"Error processing {file_path}: {e}")
        return False

def main():
    """Main function to process all TypeScript/TSX files"""
    # Get all TypeScript and TSX files
    patterns = [
        'app/**/*.tsx',
        'app/**/*.ts',
        '*.tsx',
        '*.ts'
    ]
    
    files_processed = 0
    files_fixed = 0
    
    for pattern in patterns:
        for file_path in glob.glob(pattern, recursive=True):
            if os.path.isfile(file_path):
                files_processed += 1
                if process_file(file_path):
                    files_fixed += 1
    
    print(f"\nProcessed {files_processed} files")
    print(f"Fixed {files_fixed} files")

if __name__ == "__main__":
    main()

File: fix_errors.js
<<<<<<< HEAD
const fs = require('fs');
const path = require('path');

// Function to fix unescaped entities
function fixUnescapedEntities(content) {
  // Fix specific patterns
  content = content.replace(/Let's/g, 'Let&apos;s');
  content = content.replace(/We're/g, 'We&apos;re');
  content = content.replace(/don't/g, 'don&apos;t');
  content = content.replace(/can't/g, 'can&apos;t');
  content = content.replace(/won't/g, 'won&apos;t');
  content = content.replace(/it's/g, 'it&apos;s');
  content = content.replace(/you're/g, 'you&apos;re');
  content = content.replace(/they're/g, 'they&apos;re');
  content = content.replace(/I'm/g, 'I&apos;m');
  content = content.replace(/I'll/g, 'I&apos;ll');
  content = content.replace(/I've/g, 'I&apos;ve');
  content = content.replace(/I'd/g, 'I&apos;d');
  
  return content;
}

// Function to fix missing imports
function fixMissingImports(content) {
  const commonIcons = [
    'Send', 'BarChart3', 'Shield', 'DollarSign', 'Clock', 'Star', 'Mail', 'CheckCircle',
    'Zap', 'TrendingUp', 'Brain', 'Target', 'Globe', 'Database', 'Users', 'Settings',
    'ChevronLeft', 'ChevronRight', 'ArrowRight', 'Activity', 'Lock', 'FileText',
    'CreditCard', 'Sparkles', 'PieChart', 'Cloud', 'BarChart'
  ];
  
  // Check if lucide-react is imported
  if (content.includes("from 'lucide-react'")) {
    const importMatch = content.match(/import\s*{\s*([^}]+)\s*}\s*from\s*['"]lucide-react['"]/);
    if (importMatch) {
      const existingImports = importMatch[1].split(',').map(imp => imp.trim());
      const missingImports = [];
      
      // Find missing icons in the content
      commonIcons.forEach(icon => {
        if (content.includes(`<${icon}`) && !existingImports.includes(icon)) {
          missingImports.push(icon);
        }
      });
      
      if (missingImports.length > 0) {
        const newImports = [...existingImports, ...missingImports].join(', ');
        content = content.replace(
          /import\s*{\s*[^}]+\s*}\s*from\s*['"]lucide-react['"]/,
          `import { ${newImports} } from 'lucide-react'`
        );
      }
    }
  }
  
  return content;
}

// Function to fix merge conflict markers
function fixMergeConflicts(content) {
  // Remove merge conflict markers and keep the HEAD version
  content = content.replace(/  content = content.replace(/[\s\S]*?  
  return content;
}

// Main function to process files
function processFile(filePath) {
  try {
    let content = fs.readFileSync(filePath, 'utf8');
    let modified = false;
    
    const originalContent = content;
    
    // Apply fixes
    content = fixMergeConflicts(content);
    content = fixUnescapedEntities(content);
    content = fixMissingImports(content);
    
    if (content !== originalContent) {
      fs.writeFileSync(filePath, content, 'utf8');
      console.log(`Fixed: ${filePath}`);
      modified = true;
    }
    
    return modified;
  } catch (error) {
    console.error(`Error processing ${filePath}:`, error.message);
    return false;
  }
}

// Find all TypeScript/TSX files
function findTsxFiles(dir) {
  const files = [];
  
  function traverse(currentDir) {
    const items = fs.readdirSync(currentDir);
    
    for (const item of items) {
      const fullPath = path.join(currentDir, item);
      const stat = fs.statSync(fullPath);
      
      if (stat.isDirectory() && !item.startsWith('.') && item !== 'node_modules') {
        traverse(fullPath);
      } else if (item.endsWith('.tsx') || item.endsWith('.ts')) {
        files.push(fullPath);
      }
    }
  }
  
  traverse(dir);
  return files;
}

// Process all files
const files = findTsxFiles('./app');
let fixedCount = 0;

console.log('Starting error fixes...');

files.forEach(file => {
  if (processFile(file)) {
    fixedCount++;
  }
});

console.log(`Fixed ${fixedCount} files`);
=======
const fs = require('fs');'const path = require('path');'
// Function to fix merge conflicts and syntax errors
function fixFile(filePath) {
  try {
    let content = fs.readFileSync(filePath, 'utf8');'    let fixed = false;

    // Remove merge conflict markers
    if (content.includes('      content = content.replace(/      content = content.replace(/      content = content.replace(/      fixed = true;'    }

    // Fix common syntax errors
    content = content.replace(/\(\s*\)\s*;/g, '();');'    content = content.replace(/\{\s*\}\s*;/g, '{};');'    content = content.replace(/\(\s*\)\s*\)/g, '())');'    content = content.replace(/\{\s*\}\s*\)/g, '{});;');'    
    // Fix malformed JSX
    content = content.replace(/<div\s*$/gm, '<div>');'    content = content.replace(/<div\s*>\s*$/gm, '<div>');'    
    // Fix unterminated strings
    content = content.replace(/"([^"]*?)\n/g, '"$1"');"    content = content.replace(/'([^']*?)\n/g, "'$1'");"    '    // Fix malformed template literals
    content = content.replace(/`([^`]*?)\n/g, '`$1`');'    `    // Fix missing closing tags
    const openDivs = (content.match(/<div[^>]*>/g) || []).length;
    const closeDivs = (content.match(/<\/div>/g) || []).length;
    if (openDivs > closeDivs) {
      const missingDivs = openDivs - closeDivs;
      content += '</div>'.repeat(missingDivs);'      fixed = true;
    }

    if (fixed) {
      fs.writeFileSync(filePath, content);
      console.log(`Fixed: ${filePath}`);`    }
  } catch (error) {
    console.error(`Error fixing ${filePath}:`, error.message);`  }
}

// Function to recursively find and fix files
function fixDirectory(dirPath) {
  const files = fs.readdirSync(dirPath);
  
  for (const file of files) {
    const filePath = path.join(dirPath, file);
    const stat = fs.statSync(filePath);
    
    if (stat.isDirectory() && !file.startsWith('.') && file !== 'node_modules') {'      fixDirectory(filePath);
    } else if (file.endsWith('.tsx') || file.endsWith('.ts') || file.endsWith('.jsx') || file.endsWith('.js')) {'      fixFile(filePath);
    }
  }
}

// Start fixing from the current directory
console.log('Starting error fixes...');'fixDirectory('.');'console.log('Error fixes completed!');</div></div></div>
>>>>>>> origin/main
